examples/autograd/train.py

```diff
@@ -0,0 +1,257 @@
+    # Partition training data
+    data_per_rank = train_x.shape[0] // num_workers
+    idx_start = worker_id * data_per_rank
+    idx_end = (worker_id + 1) * data_per_rank
+    train_x = train_x[idx_start:idx_end]
+    train_y = train_y[idx_start:idx_end]
+    # Partition evaluation data
+    data_per_rank = val_x.shape[0] // num_workers
+    idx_start = worker_id * data_per_rank
+    idx_end = (worker_id + 1) * data_per_rank
+    val_x = val_x[idx_start:idx_end]
+    val_y = val_y[idx_start:idx_end]
+    return train_x, train_y, val_x, val_y
+
+
+# Function to all reduce NUMPY Accuracy and Loss from Multiple Devices
+def reduce_variable(variable, dist_opt, reducer):
+    reducer.copy_from_numpy(variable)
+    dist_opt.all_reduce(reducer.data)
+    dist_opt.wait()
+    output = tensor.to_numpy(reducer)
+    return output
+
+
+def resize_dataset(x, image_size):
+    num_data = x.shape[0]
+    dim = x.shape[1]
+    X = np.zeros(shape=(num_data, dim, image_size, image_size), dtype=np.float32)
+    for n in range(0, num_data):
+        for d in range(0, dim):
+            X[n, d, :, :] = np.array(Image.fromarray(x[n, d, :, :]).resize(
+                (image_size, image_size), Image.BILINEAR),
+                                     dtype=np.float32)
+    return X
+
+
+def run(worker_id, num_workers, device_id, max_epoch, batch_size, model, data,
+        sgd):
+    dev = device.create_cuda_gpu_on(device_id)
+    dev.SetRandSeed(0)
+    np.random.seed(0)
+
+    if model == 'resnet':
+        from model import resnet
+        model = resnet.resnet50()
+    elif model == 'xceptionnet':
+        from model import xceptionnet
+        model = xceptionnet.create_model()
+    elif model == 'cnn':
+        from model import cnn
+        model = cnn.create_model()
+
+    if data == 'cifar10':
+        from data import cifar10
+        train_x, train_y, val_x, val_y = cifar10.load()
+    elif data == 'mnist':
+        from data import mnist
+        train_x, train_y, val_x, val_y = mnist.load()
+
+    if hasattr(sgd, "communicator"):
+        DIST = True
+    else:
+        DIST = False
+
+    if DIST:
+        train_x, train_y, val_x, val_y = partition(worker_id, num_workers,
+                                                   train_x, train_y, val_x,
+                                                   val_y)
+    '''
+    # check dataset shape correctness
+    if worker_id == 0:
+        print("Check the shape of dataset:")
+        print(train_x.shape)
+        print(train_y.shape)
+    '''
+
+    num_channels = train_x.shape[1]
+    image_size = train_x.shape[2]
+
+    tx = tensor.Tensor(
+        (batch_size, num_channels, model.input_size, model.input_size), dev,
+        tensor.float32)
+    ty = tensor.Tensor((batch_size,), dev, tensor.int32)
+    num_train_batch = train_x.shape[0] // batch_size
+    num_val_batch = val_x.shape[0] // batch_size
+    idx = np.arange(train_x.shape[0], dtype=np.int32)
+
+    # attached model to graph
+    graph = True
```

Comment:
the graph variable is used once. 
therefore, there is no need to define it. just pass True to `model.graph`

Reply:
Yes, I see